lib/adapters/base.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+from typing import TypedDict
+from lib.strategy import Strategy
+from lib.account import Account
+
+
+class InvalidPayloadRecieved(Exception):
+    """InvalidPayloadRecieved is thrown when an invalid message is recieved
+       from the exchange via a websocket stream."""
+    pass
+
+class Credentials(TypedDict):
+    """Required credentials to authenticate with a given exchange."""
+    exchange: str
+    key: str
+    secret: str
+
+
+class Adapter(ABC):
+    """Required interface that an exchange adapter must implement."""
```

Comment:
Do we need additional methods for the closure of connections; i.e. `close(enum)` where `enum` represents either a *user stream* or *market stream*? It would make handling OS signals gracefully easier?